apps/root/src/hooks/useOpenConnectModal.ts

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import useUser from './useUser';
+import { UserStatus } from 'common-types';
+import { defineMessage, useIntl } from 'react-intl';
+
+const useOpenConnectModal = () => {
+  const { openConnectModal } = useConnectModal();
+  const user = useUser();
+  const intl = useIntl();
+
+  return React.useCallback(() => {
+    if (!openConnectModal) return;
+
+    openConnectModal();
+
+    setTimeout(() => {
+      // Change the theme as much as we can;
+
+      // Main container
+      const container = document.querySelector(
+        'div.iekbcc0.ju367v84.ju367v6d.ju367v6y.ju367v7j > div.iekbcc0.ju367va.ju367v14'
+      );
+
+      if (container) {
+        container.className = `${container.className} rainBowKitContainer`;
+      }
+
+      const textsContainer = document.querySelector('div.iekbcc0.ju367v4.ju367vf0.ju367va.ju367v15.ju367v2m.ju367v2s');
+
+      if (textsContainer) {
+        textsContainer.className = `${textsContainer.className} rainBowKitTextsContainer`;
+        let loginText = defineMessage({
+          defaultMessage: 'Log in with your wallet',
+          description: 'RainbowLogInWithYourWallet',
+        });
+        let loginSubText = defineMessage({
+          defaultMessage:
+            "Log in with the first wallet you registered to access your account, including any other wallets and contacts you've added.",
+          description: 'RainbowLogInWithYourWalletSubText',
+        });
+        let loginDisclaimer = defineMessage({
+          defaultMessage: 'Logging in with a different wallet creates a new account without your existing data.',
+          description: 'RainbowLogInDisclaimer',
+        });
+        let loginWalletTitle = defineMessage({
+          defaultMessage: 'Select your Wallet to Log In.',
+          description: 'RainbowWalletTitle',
+        });
+
+        if (user?.status === UserStatus.loggedIn) {
+          loginText = defineMessage({
+            defaultMessage: 'Link a new wallet',
+            description: 'RainbowLinkANewWallet',
+          });
+          loginSubText = defineMessage({
+            defaultMessage: 'Connect another wallet to your already logged-in wallet',
+            description: 'RainbowLinkANewWalletSubText',
+          });
+          loginDisclaimer = defineMessage({
+            defaultMessage: 'Linking wallets allows you to track more than one!',
+            description: 'RainbowLinkWalletDisclaimer',
+          });
+          loginWalletTitle = defineMessage({
+            defaultMessage: 'Select your Wallet to link it.',
+            description: 'RainbowWalletTitle',
+          });
+        }
+
+        const titleContainer = document.querySelector('[data-testid="rk-connect-header-label"]');
+
+        if (titleContainer) {
+          titleContainer.innerHTML = intl.formatMessage(loginWalletTitle);
+        }
+
+        textsContainer.innerHTML = `
+          <div class="rainBowKitTextsTitleContainer">
```

Comment:
We are not handling both theme. Color for texts and logo should be conditioned by it